### DISCUSSION TITLE
[Azure DevOps] Renovate hangs on one particular repository

### DISCUSSION THREAD

**Discussion by mc0re** [renovatebot/renovate#17121] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

26.10.0

### Please select which platform you are using if self-hosting.

Azure DevOps Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

2020, I guess

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

It works for a good number of repositories, but for one particular it does not. I cannot figure out what makes it hang. I even installed `RE2`, which took some effort - no change.

Here is the output for that project:
```
> npx --userconfig config.npmrc -p re2@1.17.7 -p renovate@26.10.0 renovate
 INFO: Repository started (repository=[Redacted].Templates)
       "renovateVersion": "26.10.0"
 INFO: git fetch completed (repository=[Redacted].Templates)
       "durationMs": 5319
 WARN: findIssue() is not implemented (repository=[Redacted].Templates)
 INFO: Dependency extraction complete (repository=[Redacted].Templates)
       "baseBranch": "master",
       "stats": {
         "managers": {"nuget": {"fileCount": 6, "depCount": 19}},
         "total": {"fileCount": 6, "depCount": 19}
       }
```
Then it hangs - for at least 12 hours.
I know it's hard to say much without a repro, but is there any sort of an possibly eternal loop or unlimited waiting for an HTTP request? The Internet access is very limited from the corporate machines.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by mc0re** [comment]
It'd be nice to run Renovate on a local directory to try things out without actually breaking the real code.

**Comment by rarkins** [comment]
You need to first upgrade Renovate to the last version - we don't support such old releases

**Reply by mc0re** [reply]
Fair point. Unfortunately, Azure DevOps Server comes with an older GIT version, which is not supported by a newer Renovate.

**Reply by viceice** [reply]
that's only partial true. we only depend on client side git version, not server side.

**Reply by mc0re** [reply]
What do you mean? Renovate runs as an executable on a machine, in this case - Azure agent. If this agent has a wrong Git version, Renovate cannot run.

**Reply by viceice** [reply]
then you should update git on the agent in the fly. if that's not possible, run it as docker container.

we don't support such old renovate version.

**Comment by mc0re** [comment]
Ran Renovate from a Docker image (version 32.159.2), the result is the same - hangs after "Dependency extraction complete" message.

**Reply by viceice** [reply]
debug logs?

**Reply by mc0re** [reply]
If you refer to the output on the console - there is nothing more than what I posted in the first message. Yes, `LOG_LEVEL=debug`.

**Reply by rarkins** [reply]
Something is going wrong because your first message does not include any debug logs

**Reply by mc0re** [reply]
That's because I cut the log output down to what I thought was characteristical. Here is the full log from the start (attached).
[renovate-log.txt](https://github.com/renovatebot/renovate/files/9348597/renovate-log.txt)

**Reply by rarkins** [reply]
If there's more, avoid saying "there is nothing more".

**Reply by mc0re** [reply]
Sorry. What I meant was "there is nothing following those lines".

**Comment by viceice** [comment]
Are you using any kind of http proxy ? then you need to make renovate aware of it, otherwise this can cause blocked http requests

**Reply by mc0re** [reply]
AFAIK there is no proxy towards the internal sites. External sites are not available.
Note, that Renovate runs fine for a bunch of projects configured identically. In fact, I extracted this `Templates` project from the common setup, because this is the only one that fails - the rest runs smooth.

The difference between this project and the rest is that it produces a _dotnet_ template package. Thus it has a `.csproj` for the `Templates` itself and a `.csproj` that is put into the template.

**Reply by viceice** [reply]
so renovate probably hangs on trying to reach an external site, please run with `trace` log level, maybe there is something more to see.
warning, log will be huge, maybe use renovate log file and log file level trace.

**Reply by mc0re** [reply]
Here is the log of what happens after the "Dependency extraction complete" message.
[renovate.txt](https://github.com/renovatebot/renovate/files/9349443/renovate.txt)

**Reply by mc0re** [reply]
Did the log help?

**Reply by viceice** [reply]
it seems like it hungs with requests to `http://artifactory.mycompany.com/artifactory/...`, so check the access from the renovate runner machine.

eg do:

```sh
> curl -v -u user:pw 'http://artifactory.mycompany.com/artifactory/api/nuget/v3/nuget-1/FindPackagesById()?id=%27Serilog.Settings.Configuration%27&$select=Version,IsLatestVersion,ProjectUrl,Published'
```

**Reply by mc0re** [reply]
Thanks for helping. It was about accessing a certain Nuget package from our Nuget mirror repository.
